src/utils/payout.ts

```diff
@@ -0,0 +1,79 @@
+import { abi as ERC20 } from "@openzeppelin/contracts/build/contracts/ERC20.json";
+import EthersAdapter from "@gnosis.pm/safe-ethers-lib";
+import Safe from "@gnosis.pm/safe-core-sdk";
+import { SafeTransactionDataPartial } from "@gnosis.pm/safe-core-sdk-types";
+import * as ethers from "ethers";
+
+export type PayoutParams = {
+  network?: number;
+  from: string;
+  amount: string;
+  token: string; // pass 0x00 to send ether
+  to: string;
+  execute?: boolean;
+  gwei?: number;
+};
+
+export const providerKeyByChainId = new Map<number, string>([
+  [1, process.env.RPC1_ALCHEMY_KEY as string],
+  [5, process.env.RPC5_ALCHEMY_KEY as string], // goerli - safe compatable
+]);
+
+export const payout = async (args: PayoutParams) => {
+  // collect baseline info
+  const tokenAddress = ethers.utils.getAddress(args.token);
+  const isEth = tokenAddress === ethers.constants.AddressZero;
+  const destinationAddress = ethers.utils.getAddress(args.to);
+  const privateKey = process.env.PRIVATE_KEY as string;
+  const network = args.network || 1;
+  const providerKey = providerKeyByChainId.get(network);
+  if (!providerKey) {
+    throw new Error(`chainId not supported: ${network}`);
+  }
+  const provider = new ethers.providers.AlchemyProvider(network, providerKey);
+  const erc20 = new ethers.Contract(tokenAddress, ERC20, provider);
+  const decimals = isEth ? 18 : await erc20.decimals();
+  // pass x.0 if using an int otherwise shortened version will be provided
+  const amount = args.amount.split(".").length === 1 ? ethers.BigNumber.from(args.amount) : ethers.utils.parseUnits(args.amount, decimals);
```

Comment:
this line is the biggest question mark to me. because i don't know who will be calling this - besides a hardhat task, i do not know what assumptions i can make, so i was fairly defensive, but perhaps this would be better as a required boolean flag (--float=false/true)

Reply:
either that, or just requiring ints